calculator.rb

```diff
@@ -0,0 +1,101 @@
+puts "HELLO I AM CALCU-BOT. \nSELECT A MATH OPERATION."
+
+operands = [
+{id: 1, name: "ADDITION", symbol: "+"},
+{id: 2, name: "SUBTRACTION", symbol: "-"},
+{id: 3, name: "MULTIPLICATION", symbol: "*"},
+{id: 4, name: "DIVISION", symbol: "/"},
+{id: 5, name: "EXPONENTIATION", symbol: "^"},
+{id: 6, name: "MODULO", symbol: "%"}
+]
+
+operands.each do |operand|
+  puts "#{operand[:id]}. #{operand[:name]} #{operand[:symbol]}"
+end
+
+# will be assigned user-inputted operator
+math_operand = "none"
+
+while math_operand == "none"
+  user_operand = gets.chomp.upcase
+  
+  # truncate user_operand for use in case block
+  # accounts for if a user inputs "divide" or "division" (both begin with DIV)
+  user_operand = user_operand[0..2]
+  
+  case user_operand
+  when "1", "ADD", "+"
+    puts "YOU HAVE CHOSEN ADDITION."
+    math_operand = :+
+  when "2", "SUB", "-"
+    puts "YOU HAVE CHOSEN SUBTRACTION."
+    math_operand = :-
+  when "3", "MUL", "*"
+    puts "YOU HAVE CHOSEN MULTIPLICATION."
+    math_operand = :*
+  when "4", "DIV", "/"
+    puts "YOU HAVE CHOSEN DIVISION."
+    math_operand = :/
+  when "5", "EXP", "^"
+    puts "YOU HAVE CHOSEN EXPONENTIATION."
+    math_operand = :**
+  when "6", "MOD", "%"
+    puts "YOU HAVE CHOSEN MODULO-IFICATION. \n(CALCU-BOT DOES NOT KNOW THE VERB TENSE FOR A MODULO.)"
+    math_operand = :%
+  else
+    puts "CALCU-BOT HAS ENCOUNTERED AN ERROR. SELECT A MATH OPERATION."
+    math_operand = "none"
+  end
+end
+
+# ensure a value is an integer or float
+class String
+  def integer_checker
+    self.to_i.to_s == self
+  end
+  def float_checker
+    self.to_f.to_s == self
+  end
+end
```

Comment:
THIS IS COOL! And fancy! You are doing a thing which is adding a new method to the `String` class, which is what the user input it. We will definitely get into these details in a few weeks.

If you were curious about how to do this without working with the `String` class, I would likely make the methods `integer_checker` and `float_checker` not part of `String`, and define that it takes in a number, and replacing the `self` with that number, so:

```ruby
def integer_checker(num)
  return num.to_i.to_s == num
end
```

Not a mandatory refactoring, but just wanted to give you this FYI!